src/vm/olevariant.cpp

```diff
@@ -2310,8 +2310,11 @@ void OleVariant::MarshalLPSTRRArrayComToOle(BASEARRAYREF *pComArray, void *oleAr
                 ThrowOutOfMemory();
 
             // Convert the unicode string to an ansi string.
-            InternalWideToAnsi(stringRef->GetBuffer(), Length, lpstr, allocLength, fBestFitMapping, fThrowOnUnmappableChar);
-            lpstr[Length] = 0;
+            int bytesWritten = InternalWideToAnsi(stringRef->GetBuffer(), Length, lpstr, allocLength, fBestFitMapping, fThrowOnUnmappableChar);
```

Comment:
Since, allocLength = Length * GetMaxDBCSCharByteSize() + 1, can we just assert that bytesWritten < allocLength ? It would be bad if more bytes than allocLength are written.

Reply:
Certainly.  The case where bytesWritten is less than allocLength is actually quite often consider: Unicode "test" (8 bytes) will shrink to "test" (4 bytes).  Are you suggesting that I add the assert or add the assert and change the if logic as well?

Reply:
All I'm saying is that `bytesWritten` must not be larger than `allocLength` (assert if necessary). 
So why don't we just write `lpstr[bytesWritten] = 0;` instead of the `if` logic?

Reply:
Ah, great point.  I will push an update.